[quant] Support clone for per channel affine quantized tensor

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44573 [quant] Support clone for per channel affine quantized tensor**

Summary:
fixes: https://github.com/pytorch/pytorch/issues/33309

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23663828](https://our.internmc.facebook.com/intern/diff/D23663828)

jerryzh168 committed:
[quant] Support clone for per channel affine quantized tensor

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Support clone for per channel affine quantized tensor"


Summary:
fixes: https://github.com/pytorch/pytorch/issues/33309

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on aten/src/ATen/native/quantized/QTensor.cpp:
```diff
@@ -153,12 +148,26 @@ Tensor quantized_clone(
+        self.q_per_channel_zero_points(),
+        self.q_per_channel_axis(),
+        self.options().memory_format(memory_format),
+        c10::nullopt);
```
@VitalyFedyunin is the memory format use correct? https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L1367

VitalyFedyunin replied:
Yes, it looks like correct.

VitalyFedyunin reviewed (approved):
Memory format usage is correct

jerryzh168 committed:
Update on "[quant] Support clone for per channel affine quantized tensor"


Summary:
fixes: https://github.com/pytorch/pytorch/issues/33309

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23663828](https://our.internmc.facebook.com/intern/diff/D23663828)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Support clone for per channel affine quantized tensor"


Summary:
fixes: https://github.com/pytorch/pytorch/issues/33309

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23663828](https://our.internmc.facebook.com/intern/diff/D23663828)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Support clone for per channel affine quantized tensor"


Summary:
fixes: https://github.com/pytorch/pytorch/issues/33309

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23663828](https://our.internmc.facebook.com/intern/diff/D23663828)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Support clone for per channel affine quantized tensor"


Summary:
fixes: https://github.com/pytorch/pytorch/issues/33309

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23663828](https://our.internmc.facebook.com/intern/diff/D23663828)

[ghstack-poisoned]
